src/main/java/com/PBA/authservice/facade/UserFacadeImpl.java

```diff
@@ -0,0 +1,84 @@
+import com.pba.authservice.persistance.model.PendingUserProfile;
+import com.pba.authservice.persistance.model.dtos.UserDto;
+import com.pba.authservice.persistance.model.dtos.UserProfileDto;
+import com.pba.authservice.service.ActiveUserService;
+import com.pba.authservice.service.EmailService;
+import com.pba.authservice.service.PendingUserService;
+import com.pba.authservice.controller.request.UserCreateRequest;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.UUID;
+
+@Component
+public class UserFacadeImpl implements UserFacade {
+    private final PendingUserService pendingUserService;
+    private final PendingUserMapper pendingUserMapper;
+    private final ActiveUserService activeUserService;
+    private final ActiveUserMapper activeUserMapper;
+    private final EmailService emailService;
+
+    public UserFacadeImpl(PendingUserService pendingUserService, PendingUserMapper pendingUserMapper, ActiveUserService activeUserService, ActiveUserMapper activeUserMapper, EmailService emailService) {
+        this.pendingUserService = pendingUserService;
+        this.pendingUserMapper = pendingUserMapper;
+        this.activeUserService = activeUserService;
+        this.activeUserMapper = activeUserMapper;
+        this.emailService = emailService;
+    }
+
+    @Override
+    public void registerUser(UserCreateRequest userCreateRequest) {
+        PendingUser pendingUser = pendingUserMapper.toPendingUser(userCreateRequest);
+        PendingUser savedPendingUser = pendingUserService.addPendingUser(pendingUser);
+
+        PendingUserProfile pendingUserProfile = pendingUserMapper.toPendingUserProfile(userCreateRequest, savedPendingUser.getId());
+        pendingUserService.addPendingUserProfile(pendingUserProfile);
+
+        emailService.sendVerificationEmail(userCreateRequest.getEmail(), savedPendingUser.getValidationCode());
+    }
+
+    @Override
+    public UserDto getUser(UUID uid) {
+        ActiveUser activeUser = activeUserService.getUserByUid(uid);
+        ActiveUserProfile activeUserProfile = activeUserService.getProfileByUserId(activeUser.getId());
+
+        UserProfileDto userProfileDto = activeUserMapper.toUserProfileDto(activeUserProfile);
+        return activeUserMapper.toUserDto(activeUser, userProfileDto);
+    }
+
+    @Override
+    @Transactional
+    public UserDto verifyUser(UUID validationCode) {
+        PendingUser userToValidate = pendingUserService.getPendingUserByValidationCode(validationCode);
+        this.validatePendingUser(userToValidate);
+        ActiveUser activeUser = pendingUserMapper.toActiveUser(userToValidate);
+        PendingUserProfile pendingUserProfile = pendingUserService.getPendingUserProfileByUserId(userToValidate.getId());
+        this.deletePendingUser(userToValidate, pendingUserProfile);
+
+        ActiveUser savedActiveUser = activeUserService.addUser(activeUser);
```

Comment:
```suggestion
        addActiveUser(userToValidate)
```

You have 4 lines that can be extracted to one function. And it's related to creating an active user,